google-built-opentelemetry-collector/Makefile

```diff
@@ -132,6 +138,17 @@ clean-collector: clean-build-dir
 clean-build-dir:
 	rm -rf $(COLLECTOR_BUILD_TEMP_DIR)
 
+###################
+# Goreleaser build
+###################
+# These targets are for invoking goreleaser.
+
+.PHONY: goreleaser-release
```

Comment:
Is this the main target going to be used by the Louhi flow? Asking as this will build all architectures + OSes at the same time. 

With CGO_ENABLED on, I remember we have to build on the corresponding architecture and os (can only build arm in arm images). Do we need to add a target to build a single arch+os at a time?

Reply:
Yeah with `CGO_ENABLED=1` goreleaser actually kind of falls apart in general. The operation of it is based on the idea that the cross-compile will work all at once. Building a single target out of a goreleaser config file is a pro-only feature, so we won't be able to leverage it.

I'm thinking as a future improvement that I'm going to add a concept to distrogen of toggle-able "Distribution Features", and will make `goreleaser` the first option. This way, I can produce an error when trying to produce a distribution with `cgo_enabled` and `goreleaser` feature on, since that's not possible. Then I will turn off `goreleaser` functionality in `otelopscol` distribution, where we are not using it anyway.

I want to do that in a followup PR, since that will be a bit of a larger diff.